util/syspolicy/setting/policy_scope.go

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package setting
+
+import (
+	"fmt"
+	"strings"
+
+	"tailscale.com/types/lazy"
+)
+
+var (
+	lazyCurrentScope lazy.SyncValue[PolicyScope]
+
+	// DeviceScope indicates a scope containing device-global policies.
+	DeviceScope = PolicyScope{kind: DeviceSetting}
+	// CurrentProfileScope indicates a scope containing policies that apply to the
+	// currently active Tailscale profile.
+	CurrentProfileScope = PolicyScope{kind: ProfileSetting}
+	// CurrentUserScope indicates a scope containing policies that apply to the
+	// current user, for whatever that means on the current platform and
+	// in the current application context.
+	CurrentUserScope = PolicyScope{kind: UserSetting}
+)
+
+// PolicyScope is a management scope.
+type PolicyScope struct {
+	kind      Scope
+	userID    string
+	profileID string
+}
+
+// DefaultScope returns the default [PolicyScope] to be used by a program.
+// It returns [DeviceScope], unless explicitly changed with [SetCurrentScope].
+func DefaultScope() PolicyScope {
+	return lazyCurrentScope.Get(func() PolicyScope { return DeviceScope })
+}
+
+// SetCurrentScope attempts to set the specified scope as the default scope.
+// It fails and returns false if called more than once, or if the [DefaultScope]
+// has already been used.
+func SetCurrentScope(scope PolicyScope) bool {
```

Comment:
The idea is that there are policy _settings_ (individual configurable items, if you prefer) and policies, which represent a set of policy _settings_ that apply to a specific policy scope.

Policy settings have `setting.Scope`s, indicating whether they manage device-global, profile-specific, or user-specific resources/state/behavior.

`PolicyScope`, however, represents a specific scope for which a policy might be configured, or a resultant policy might be obtained.
Therefore, it may include a profile and/or user ID.

The `SetCurrentScope` function changes the scope for which the `syspolicy/*` packages return resultant policy (or individual policy settings) unless a different scope is explicitly requested (as documented for the `CurrentScope` accessor).

I chose `CurrentScope` as the name because I thought it better reflected the idea that it would eventually include policy settings that apply to the current Tailscale profile, and also its kind of analogous to the `HKEY_CURRENT_USER` registry hive on Windows.

However, on second thought, I think renaming it to `DefaultScope` might make more sense. Is there anything else I should do to make this whole idea more clear?

Reply:
Renamed; updated the docstring for both `DefaultScope` and `SetDefaultScope`.